### DISCUSSION TITLE
How do I make generated types use `undefined` instead of `null`

### DISCUSSION THREAD

**Discussion by alielkhateeb** [prisma/prisma#13979] [Q&A] [answered]
I'm using Postgresql, and here is the scenario I'm trying to solve:

If I have a model like that:
```
model Booking {
  resourceId           String                  @id @map("resource_id") @db.Uuid
  requestedDate    DateTime            @map("requested_delivery_date") @db.Date
  confirmedDate    DateTime?          @map("confirmed_delivery_date") @db.Date
}
```

The generated type in Prisma Client will be:
```
export type Booking = {
  resourceId: string
  requestedDate: Date
  confirmedDate: Date | null
}
```

Now in typescript, I would like to be able to consume Prisma Client types and not be forced to duplicate the types with `undefined` instead of `null`, I would like to get something like this to work (Since `confirmedDate` is optional)
```
import { Booking } from '@prisma/client';

const booking: Booking = {
    resourceId: '1',
    requestedDate: new Date()
}
```

So I am curious what did others do to solve this?

**Comment by austincrim** [comment, marked as answer]
Hey @alielkhateeb 👋,

It is not possible to modify the generated types in this way.

You could maybe create your own [mapped type](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html) based on the Prisma type as a workaround.

Would you mind opening a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) outlining your issue here? That would be really helpful!

**Reply by alielkhateeb** [reply]
Hi @austincrim thanks for the response, I created a new [feature request](https://github.com/prisma/prisma/issues/13987) :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
